app/imports/api/users/methods.js

```diff
@@ -38,3 +40,60 @@ export const removeAdmin = new ValidatedMethod({
     },
 });
 
+export const setProfileName = new ValidatedMethod({
+    name: 'users.setName',
+    validate: setNameSchema.validator({ clean: true }),
+    run({ name }) {
+        const userId = Meteor.userId();
+
+        if (userId) {
+            return Meteor.users.update(userId, {
+                $set: {
+                    'profile.name': name,
+                },
+            });
+        }
+        return undefined;
+    },
+});
+
+export const setGravatarAvatar = new ValidatedMethod({
+    name: 'users.setGravatar',
+    validate: new SimpleSchema({}).validator({}),
+    run() {
+        const user = Meteor.user();
+
+        if (user) {
+            const userId = user._id;
+            const email = user.services.github.email;
```

Comment:
1) What if user is connected by other service?
2) Remember - use public and primary email.
3) You can create additional field in user document, and it's preferred:
```js
{
    // other fields
    emails: [
        { address: "user@email", verified: true }
    ]
}
```
See how it looks in `Accounts`

In my opinion you should treat `services` as private field. If you create field like emails, it will be better - if new service will be added, eg LDAP or another oauth you can handle it without editing a lot of files in whole project.